noir-projects/noir-protocol-circuits/crates/types/src/traits.nr

```diff
@@ -155,6 +155,35 @@ impl FromField for U128 {
 }
 
 // docs:start:serialize
+/// Trait for serializing Noir types into arrays of Fields in a constraint-efficient manner.
+///
+/// This trait prioritizes minimal constraint costs during serialization, even if it
+/// results in larger array sizes. Each value is typically converted directly into one
+/// or more Fields without any packing or compression. This trait (and Deserialize) are
+/// typically used to communicate between Noir and TypeScript (via oracles and function
+/// arguments).
```

Comment:
Again this is not why Serialize exists. You *cannot* do Serialize in a different way, e.g. in a different order. Each struct only has *one* correct implementation of Serialize - if you try to make a different one (that may be e.g. more constraint efficient) it'll be wrong and things will break.

cc @Thunkar

Reply:
This is also why Serialize and Deserealize are separate (each by their own is useful because they implicitly exist in TS already, so they *do* have a counterpart), but Packable is both fns (because they are their own counterpart, there is e.g. no native `unpack`).

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/11136/commits/bbc6d17d93ba7ae9b7fd5370fd09ffd1589f8283